packages/core/src/blitz-app-root.tsx

```diff
@@ -80,19 +83,24 @@ export function withBlitzAppRoot(UserAppRoot: React.ComponentType<any>) {
       props.Component.authenticate !== undefined ||
       props.Component.redirectAuthenticatedTo
 
+    const queryClientRef = useRef<QueryClient>()
+    if (!queryClientRef.current) {
+      queryClientRef.current = queryClient
+    }
+
     useEffect(() => {
       document.documentElement.classList.add("blitz-first-render-complete")
     }, [])
 
     return (
-      <>
+      <BlitzProvider
+        client={queryClientRef.current}
+        dehydratedState={props.pageProps.dehydratedState}
+      >
```

Comment:
- [ ] Let's move this queryClient ref into BlitzProvider
- [ ] Change BlitzProvider to use default Blitz queryClient so that `queryClient` prop is optional
- [ ] Remove `queryClient` prop from app `test-utils`.

That will streamline the API for end users

Reply:
Cool got those ones in now 👌